crates/re_types_builder/src/codegen/rust.rs

```diff
@@ -2453,14 +2464,21 @@ fn quote_arrow_field_deserializer(
                     offsets
                         .enumerate()
                         .map(move |(i, (start, end))| bitmap.as_ref().map_or(true, |bitmap| bitmap.get_bit(i)).then(|| {
```

Comment:
Should we check that `start < end`, i.e. check that `data_src.offsets` is never decreasing?

Btw, does `data_src.offsets` always contain an offset to the one-past-end of the last item? (I assume so, otherwise we would miss the last range). It would be nice with a comment about that earlier up

Reply:
Those are [invariants from the Arrow specification](https://arrow.apache.org/docs/format/Columnar.html#variable-size-binary-layout). I do agree that we need to defend against non-spec compliant payloads on the deserialization path, but that's out of the scope of this PR: it'll be part of my upcoming refactor.